### ISSUE TITLE
TestReadWriteSync randomly fails

### ISSUE THREAD

**Issue by QuLogic** [creack/goselect#17]
When rebuilding for Fedora Rawhide, I'm seeing random failures of `TestReadWriteSync`. It failed on i686 once, and x86_64 twice, while passing on 4 other arches, but in another rebuild, everything passed. This is using Golang 1.16.6, and it fails with something like:
```
github.com/creack/goselect
--- FAIL: TestReadWriteSync (1.03s)
    select_test.go:51: unexpected fd ready: 40,expected: 39
```
(though the number sometimes differs)

**Comment by creack** [comment]
Should be fixed by #18

**Comment by QuLogic** [comment]
Yes, thanks, that works.
